### ISSUE TITLE
Catalyst support

### ISSUE THREAD

**Issue by NachoSoto** [Carthage/Carthage#2799]
Apparently in order for this to work Carthage will need to build `.xcframework`s now: https://twitter.com/NeoNacho/status/1135691439734575104.

**Comment by ikesyo** [comment]
`.xcframework` can be made by `xcodebuild -create-xcframework` which takes `-framework` and `-out` options. So that's means that we should run the command instead of `lipo`ing.

**Comment by blender** [comment]
example:
```
$ xcodebuild archive -workspace Alamofire.xcworkspace -scheme "Alamofire iOS" -sdk iphoneos13.0 OBJROOT=build/iOS
$ xcodebuild archive -workspace Alamofire.xcworkspace -scheme "Alamofire iOS" -sdk iphonesimulator13.0 OBJROOT=build/simulator
$ xcodebuild -create-xcframework -framework build/iOS/UninstalledProducts/iphoneos/Alamofire.framework -framework build/simulator/UninstalledProducts/iphonesimulator/Alamofire.framework -output build/Alamofire.xcframework
```

**Comment by blender** [comment]
Can anyone actually build Carthage on Xcode 11 Swift 5.1?

**Comment by alexaubry** [comment]
@blender I just tried, it seems unhappy with the currying in the `OptionsProtocol` objects evaluation (e.g. in `CheckoutCommand.Options.evaluate`):

![Error in ](https://user-images.githubusercontent.com/16192914/58887547-1d149900-86e6-11e9-9dbf-dec6c5926c13.png)

Using `Options.init` seems to work, though.

**Comment by blender** [comment]
Yeah, I was trying to go around this by specifying a lower version of swift in the Makefile but I'm hitting a wall.

**Comment by ikesyo** [comment]
Looks like building Carthage itself with Xcode 11 (Swift 5.1) is a separate issue.

**Comment by blender** [comment]
I’m on this btw ;)

**Comment by blender** [comment]
https://github.com/Carthage/Carthage/pull/2801 this is in progress but can already build xcframeworks with `carthage  build --platform iOS --no-skip-current --create-xcframework`

I'm not sure what we're supposed to do with dSYMs

**Comment by haithngn** [comment]
> Can anyone actually build Carthage on Xcode 11 Swift 5.1?

yes I did but no luck,
It seems the framework will be build in swift 4.1.2, that is very strange.

**Comment by blender** [comment]
@haithngn I believe this is a problem with your setup. Carthage itself is both building correctly Xcode 11 Swift 5.1 and so frameworks built by Carthage that support Swift 5.1

**Comment by haithngn** [comment]
@blender , you right :D I have tried on xcode 9.4 then the IDE say "module compiled with Swift 5.1 cannot be imported in Swift 4.1.2" 😂
I will try on xcode 11 later

**Comment by mdiep** [comment]
This should mean that we can stop using `copy-frameworks` then?

**Comment by blender** [comment]
@mdiep as far as I can tell yes. But I'm not sure how dsyms will be packaged.

**Comment by jaelee1993** [comment]
Hey guys, I was just wondering was there a resolution to this problem?

**Comment by tobyyoungberg** [comment]
It looks like dSYM files are inside each architecture package.  `ios.xcarchive/dSYMs`  Can they simply be copied next to the XCFramework or something?  Is there a way to package them inside the xcframework as part of the `create-xcframework` command?

**Comment by LamineNdy** [comment]
Hello, any ETA planned for this feature?

**Comment by steipete** [comment]
CocoaPods recently added support for xcframeworks in 1.9.0, we (PSPDFKit) sponsored part of the development. Would anyone be interested in a similar agreement?

https://github.com/CocoaPods/CocoaPods/issues/9148

**Comment by DavidBrunow** [comment]
I’d be happy to implement Catalyst functionality on my employer’s time but I am worried that Carthage has no forward movement. There are multiple other pull requests that seem to be stuck.

**Comment by tmspzz** [comment]
@DavidBrunow which ones? I have merged quite a few recently.

I also have contacted @steipete since I have some time on my hands but I can't guarantee it will be enough.

**Comment by DavidBrunow** [comment]
@tmspzz Sorry I was wrong about the pull requests not being merged -- I hadn't noticed that they had been. But we are still waiting on a release for those pull requests so I'm still worried.

**Comment by paulb777** [comment]
I just tried to install a binary xcframework via Carthage. It installed the simulator .framework slice and threw out the arm .framework slice.

**Comment by mlostekk** [comment]
is carthage dead?

**Comment by fnuky** [comment]
> is carthage dead? or will a fix come one day?

Probably is :D I had to switch to spm because I can't wait anymore

**Comment by evandcoleman** [comment]
It's most certainly not dead. XCFramework support is actually now on `master` as of a few weeks ago. https://github.com/Carthage/Carthage/pull/3071

**Comment by mlostekk** [comment]
Ah wonderful! Any ETA for a public release?

**Comment by nishantd-03** [comment]
looks like the changes are released in today's 0.37 release.

**Comment by tmspzz** [comment]
`O.37.0` includes support for .xcframeworks but not yet for Catalyst

**Comment by ddaddy** [comment]
Now that we have `XCFrameworks` support, is there a way to get the Catalyst slice built?

**Comment by hansemannn** [comment]
Happy to sponsor Catalyst slice support, please feel to contact me via Twitter `[@hansemannnn](https://twitter.com/hansemannnn)´!

**Comment by NachoSoto** [comment]
I pushed a working implementation: #3235

Not quite ready to be merged yet, but if you want to try it out in your project, feel free to clone it and `make install`, and let me know if you encounter any issues!
